lib/python/picongpu/extra/input/preparingInsightData.py

```diff
@@ -0,0 +1,472 @@
+        """
+        Y, X, W = np.meshgrid(self.y, self.x, self.w)
+        fac = (
+            1j
+            / (2 * np.pi * c)
+            * W
+            / self.foc
+            * np.exp(-1j * W / c / 2 / self.foc * (X**2 + Y**2) * (1 - d / self.foc))
+        )
+        NF = np.fft.fftshift(np.fft.ifft2(np.fft.ifftshift(self.Ew * fac, axes=(0, 1)), axes=(0, 1)), axes=(0, 1))
+        a = np.fft.fftshift(np.fft.fftfreq(self.x.size, self.dx))
+        b = np.fft.fftshift(np.fft.fftfreq(self.y.size, self.dy))
+        # rescaling from spatial frequency to space for every lambda (x = a*lambda*f)
+        NF_resc = np.empty_like(NF, dtype=complex)
+        scale = 2 * np.pi * self.foc * c / self.w
+        self.x_NF = np.linspace(a[0] * scale[-1], a[-1] * scale[-1], a.size)
+        self.y_NF = np.linspace(b[0] * scale[-1], b[-1] * scale[-1], b.size)
+        Y_NF, X_NF = np.meshgrid(self.y_NF, self.x_NF, indexing="ij")
+        for i in range(self.w.size):
+            interp_NF = RegularGridInterpolator((b * scale[i], a * scale[i]), NF[:, :, i], method=method)
+            NF_resc[:, :, i] = interp_NF((Y_NF, X_NF))
+
+        self.Ew_NF = NF_resc / np.abs(NF_resc).max()  # rescale amplitude to 1
+
+    def NFtoFF(self, d=0, method="linear"):
+        """
+        Propagate near field (in frequency domain) to far field from distance d in front of lens
+
+        Arguments:
+        d: distance to lens, default is 0
+        method: interpolation method of RegularGridInterpolator, default is 'linear'
+        """
+        a = np.fft.fftshift(np.fft.fftfreq(self.x_NF.size, self.dx_NF))
+        b = np.fft.fftshift(np.fft.fftfreq(self.y_NF.size, self.dy_NF))
+        B, A, W = np.meshgrid(b, a, self.w)
+        fac = (
+            W
+            / (1j * c * 2 * np.pi * self.foc)
+            * np.exp(1j * 2 * np.pi**2 * self.foc * c / W * (A**2 + B**2) * (1 - d / self.foc))
+        )
+        FF = np.fft.fftshift(np.fft.fft2(np.fft.fftshift(self.Ew_NF, axes=(0, 1)), axes=(0, 1)), axes=(0, 1)) * fac
+        # rescaling from spatial frequency to space for every lambda
+        scale = 2 * np.pi * self.foc * c / self.w
+        self.x = np.linspace(a[0] * scale[-1], a[-1] * scale[-1], a.size)
+        self.y = np.linspace(b[0] * scale[-1], b[-1] * scale[-1], b.size)
+        Y, X = np.meshgrid(self.y, self.x, indexing="ij")
+        for i in range(self.w.size):
+            interp_FF = RegularGridInterpolator((a * scale[i], b * scale[i]), FF[:, :, i], method=method)
+            self.Ew[:, :, i] = interp_FF((Y, X))
+
+        self.Ew = self.Ew / np.abs(self.Ew).max()  # rescale amplitude to 1
+
+    def correctPhase(self, GVD=0, TOD=0):
+        """
+        Since there is no information about the global phase of every wavelength measured with Insight,
+        there has to be a phase correction.
+        For that, we assume perfect compression of the beam center in the near field, thus we substract
+        a global phase for every wavelength.
```

Comment:
", thus we subtract a frequency's phase value at the beam center from every phase value in the frequency's  transverse distribution in the NF." is more precise and correct, isn't it?